### PR TITLE
refactor(ecocredit): update core simulations to use v1 api

### PR DESCRIPTION
**Pull request by aleem1314**
## Description

Closes: #920

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Review by ryanchristo — APPROVED** [review]
This looks great. Nice work.

**Comment by technicallyty** [comment]
can we uncomment the sims runner now? or is there more to do after this?

**Comment by aleem1314** [comment]
> can we uncomment the sims runner now? or is there more to do after this?

Genesis simulations are still using `v1alpha2`. We can enable sims after we migrate genesis state to v1 api #977.

File: x/ecocredit/simulation/genesis.go [hunk 1]
```diff
@@ -52,7 +53,24 @@ func genAllowListEnabled(r *rand.Rand) bool {
...
+		{
+			Name:         "biodiversity",
+			Abbreviation: "BIO",
+			Unit:         "ton",
+			Precision:    6, // TODO: randomize precision, precision is currently locked to 6
+		},
+	}
+}
+
+func genCreditTypesLegacy(r *rand.Rand) []*ecocredit.CreditType {
```
Comment (aaronc) [inline x/ecocredit/simulation/genesis.go:73]: why do we still need this legacy function?

**Review by aaronc — APPROVED** [review]

File: x/ecocredit/simulation/genesis.go (hunk 1 above)
Comment (ryanchristo) [reply x/ecocredit/simulation/genesis.go:73]: We need to finish migrating genesis from v1alpha1 to v1 (see #977). @aleem1314 can you make sure we remove the use of legacy types in #977?

**Pull request merged**